Source/WebGPU/WebGPU/BindGroupLayout.h

```diff
@@ -114,10 +114,20 @@ class BindGroupLayout : public WGPUBindGroupLayoutImpl, public RefCounted<BindGr
     uint32_t sizeOfVertexDynamicOffsets() const;
     uint32_t sizeOfFragmentDynamicOffsets() const;
     uint32_t sizeOfComputeDynamicOffsets() const;
+    const Device& device() const;
+    bool isAutoGenerated() const;
+
+    uint32_t uniformBuffersPerStage(ShaderStage) const;
+    uint32_t storageBuffersPerStage(ShaderStage) const;
+    uint32_t samplersPerStage(ShaderStage) const;
+    uint32_t texturesPerStage(ShaderStage) const;
+    uint32_t storageTexturesPerStage(ShaderStage) const;
+    uint32_t dynamicUniformBuffers() const;
+    uint32_t dynamicStorageBuffers() const;
 
 private:
-    BindGroupLayout(StageMapTable&&, id<MTLArgumentEncoder>, id<MTLArgumentEncoder>, id<MTLArgumentEncoder>, EntriesContainer&&, size_t sizeOfVertexDynamicOffsets, size_t sizeOfFragmentDynamicOffsets, size_t sizeOfComputeDynamicOffsets);
-    explicit BindGroupLayout();
+    BindGroupLayout(StageMapTable&&, id<MTLArgumentEncoder>, id<MTLArgumentEncoder>, id<MTLArgumentEncoder>, EntriesContainer&&, size_t sizeOfVertexDynamicOffsets, size_t sizeOfFragmentDynamicOffsets, size_t sizeOfComputeDynamicOffsets, bool isAutoGenerated, ShaderStageArray<uint32_t>&& uniformBuffersPerStage, ShaderStageArray<uint32_t>&& storageBuffersPerStage, ShaderStageArray<uint32_t>&& samplersPerStage, ShaderStageArray<uint32_t>&& texturesPerStage, ShaderStageArray<uint32_t>&& storageTexturesPerStage, uint32_t dynamicUniformBuffers, uint32_t dynamicStorageBuffers, const Device&);
```

Comment:
I feel these would be better packed in a `BindGroupLayoutInit` struct.

Reply:
I agree, I filed https://bugs.webkit.org/show_bug.cgi?id=266392 for tracking